src/atlases.jl

```diff
@@ -0,0 +1,157 @@
+
+"""
+    abstract type AbstractAtlas end
+
+An abstract class for atlases.
+"""
+abstract type AbstractAtlas end
+
+"""
+    struct RetractionAtlas{
+        TInvRetr<:AbstractInverseRetractionMethod,
+        TRetr<:AbstractRetractionMethod,
+        TBasis<:AbstractBasis,
+    } <: AbstractAtlas
+
+An atlas indexed by points on a manifold, such that coordinate transformations are performed
+using retractions, inverse retractions and coordinate calculation for a given basis.
+"""
+struct RetractionAtlas{
+    TInvRetr<:AbstractInverseRetractionMethod,
+    TRetr<:AbstractRetractionMethod,
+    TBasis<:AbstractBasis{<:Any,TangentSpaceType},
+} <: AbstractAtlas
+    invretr::TInvRetr
+    retr::TRetr
+    basis::TBasis
+end
+
+function RetractionAtlas(
+    invretr::AbstractInverseRetractionMethod,
+    retr::AbstractRetractionMethod,
+)
+    return RetractionAtlas(invretr, retr, DefaultOrthonormalBasis())
+end
+RetractionAtlas() = RetractionAtlas(LogarithmicInverseRetraction(), ExponentialRetraction())
+
+get_default_atlas(M::Manifold) = RetractionAtlas()
+
+"""
+    get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
+
+Calculate coordinates of point `p` on manifold `M` in chart from atlas `A` at index `i`.
+"""
+get_point_coordinates(::Manifold, ::AbstractAtlas, ::Any, ::Any)
+
+function get_point_coordinates(M::Manifold, A::AbstractAtlas, i, p)
+    x = allocate_result(M, get_point_coordinates, p)
+    get_point_coordinates!(M, x, A, i, p)
+    return x
+end
+
+function get_point_coordinates!(M::Manifold, x, A::RetractionAtlas, i, p)
+    return get_coordinates!(M, x, i, inverse_retract(M, i, p, A.invretr), A.basis)
+end
+
+function get_point_coordinates(M::Manifold, A::RetractionAtlas, i, p)
+    return get_coordinates(M, i, inverse_retract(M, i, p, A.invretr), A.basis)
+end
+
+"""
+    get_point(M::Manifold, A::AbstractAtlas, i, x)
+
+Calculate point at coordinates `x` on manifold `M` in chart from atlas `A` at index `i`.
+"""
+get_point(::Manifold, ::AbstractAtlas, ::Any, ::Any)
+
+function get_point(M::Manifold, A::RetractionAtlas, i, x)
+    return retract(M, i, get_vector(M, i, x, A.basis), A.retr)
+end
+
+function get_point!(M::Manifold, p, A::RetractionAtlas, i, x)
+    return retract!(M, p, i, get_vector(M, i, x, A.basis), A.retr)
+end
+
+"""
+    select_chart(M::Manifold, A::AbstractAtlas, p)
+
+Select a chart from atlas `A` for manifold `M` that is suitable for representing
+neighborhood of point `p`.
+"""
+select_chart(::Manifold, ::AbstractAtlas, ::Any)
+
+select_chart(::Manifold, ::RetractionAtlas, p) = p
```

Comment:
You here return the index of the chart in the `::RetractionAtlas`, right? Would it maybe be reasonable to actually return the char, i.e. a function?

Reply:
Returning the index is more general, for example one may want to use it to do `get_point_coordinates`, `get_point` or in `transition_map`. We can always add a convenience function for that later.

Reply:
Then I would maybe change the name to `get_chart_index`? Just semantically I would `get_chart` (sure a convenience function, that could then be added after) to actually return a chart.

Reply:
Sure, I'll change the name.